File: 프로그래머스/unrated/181859. 배열 만들기 6/배열 만들기 6.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(int[] arr) {
        List<Integer> answer = new ArrayList<>();
        
        for (int i=0; i<arr.length; i++) {

                if (!answer.isEmpty() && arr[i] == answer.get(answer.size()-1)) {
                    answer.remove(answer.size()-1);  
                } else {
                    answer.add(arr[i]);   
                }
            
        }
        
        return answer.isEmpty() ? new int[] {-1} : answer.stream().mapToInt(e -> e).toArray();
    }
}

File: 프로그래머스/unrated/181859. 배열 만들기 6/배열 만들기 6.py
def solution(arr):
    answer = []
    
    for i in range(len(arr)):
        if len(answer) != 0 and answer[len(answer)-1] == arr[i]:
            answer = answer[:len(answer)-1]
        else:
            answer.append(arr[i])
    
    return answer if answer else [-1]